### DISCUSSION TITLE
Handling Large-Scale Static Generation with generateStaticParams() in Next.js v13

### DISCUSSION THREAD

**Discussion by Lashapor** [vercel/next.js#58006] [Help]
### Summary

Hello Next.js Community,

I'm encountering an issue while trying to statically generate a large number of pages in my Next.js v13 project. The goal is to create individual pages for each item in a dataset of approximately 1500 symbols, leveraging a generateStaticParams() function.

Here's what I'm experiencing:

The build process successfully generates around 1000 pages but then seems to halt, leaving around 500 pages unbuilt.
There are no explicit errors thrown, and the build just stops, suggesting some form of limit or timeout might be getting triggered.

### Additional information

```js
export async function generateStaticParams() {
  const symbols = await FetchAllSymbolsData();

  return symbols.map((etf: string) => ({
    symbol: etf.toLowerCase()
  }));
}

export default async function Home(props: pageProps) {
  let symbol = props.params.symbol.toUpperCase();

  const oneSymbolData = await FetchSymbolData(symbol);

  if (!oneSymbolData.data) {
    return notFound();
  }

  const blogData = await fetchFilteredBlogs(symbol)

  return (
    <>
      <ETFAnalysis symbol={symbol} />
      <ETFDescription symbolData={oneSymbolData.data} />
........
```

**Comment by RusseII** [comment]
idk but i'm generating 100k pages and was having error when I was console.logging and running out of log space
